pkg/util/network_docker.go

```diff
@@ -0,0 +1,28 @@
+// +build docker
+
+package util
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+	"github.com/DataDog/datadog-agent/pkg/util/docker"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// GetAgentNetworkMode retrieves from Docker the network mode of the Agent container
+func GetAgentNetworkMode() (string, error) {
+	cacheNetworkModeKey := cache.BuildAgentKey("networkMode")
+	if cacheNetworkMode, found := cache.Cache.Get(cacheNetworkModeKey); found {
+		return cacheNetworkMode.(string), nil
+	}
+
+	log.Debugf("GetAgentNetworkMode trying Docker")
+	networkMode, err := docker.GetAgentContainerNetworkMode()
+	if err != nil {
```

Comment:
Do you think we should cache the failure to avoid retrying something that will never succeed (i think hostConfig is immutable).

Reply:
makes sense 👍